### ISSUE TITLE
Training with GRPO takes longer than reported

### ISSUE THREAD

**Issue by Lineark** [huggingface/open-r1#251]
Running the given GRPO training script takes around 9 hours(estimated from early iterations, excluding time for validation) instead of the reported ~3 hours. I used 8x H100 SXM 80GB rented from cloud. Did not modify any configurations.

> Running this on 8 H100 80G GPU takes about 3 hours:
> 
> ```
> ACCELERATE_LOG_LEVEL=info accelerate launch --config_file recipes/accelerate_configs/zero2.yaml \
>     --num_processes=7 src/open_r1/grpo.py \
>     --config recipes/Qwen2.5-Math-7B/grpo/config_simple_rl.yaml
> ```

**Comment by Lineark** [comment]
>   1%|▌                                                                                       | 3/468 [03:25<8:51:40, 68.60s/it]{'loss': 0.0, 'grad_norm': 0.6876494288444519, 'learning_rate': 3.1914893617021275e-07, 'rewards/accuracy_reward': 0.5500000290572643, 'rewards/format_reward': 0.0, 'rewards/reasoning_steps_reward': 0.2690476364456117, 'rewards/cosine_scaled_reward': 0.2598649523220956, 'reward': 1.078912627696991, 'reward_std': 0.8824680134654045, 'completion_length': 639.0250267028808, 'kl': 0.00010697841644287109, 'epoch': 0.01}
>   1%|██                                                                                                                       ^[[B
> 
>   2%|█▋                                                                                      | 9/468 [10:25<8:55:34, 70.01s/it]

I continued to run this for a little while and the estimated time did not change much.

But it was not a complete run, it's possible that the completion_length later drops and causes a reduction in required time.
